core/embed/rust/src/ui/component/swipe_detect.rs

```diff
@@ -174,20 +176,30 @@ impl SwipeDetect {
         }
     }
 
-    const fn min_lock(&self) -> u16 {
-        Self::MIN_LOCK
+    const fn min_lock(&self, dir: SwipeDirection) -> u16 {
+        match dir {
+            SwipeDirection::Up | SwipeDirection::Down => Self::MIN_LOCK,
+            SwipeDirection::Left | SwipeDirection::Right => {
+                Self::MIN_LOCK * Self::VERTICAL_PREFERENCE as u16
```

Comment:
What is the precedence of operations here?

Is

``` rust
Self::MIN_LOCK * Self::VERTICAL_PREFERENCE as u16
```

the same as

``` rust
(Self::MIN_LOCK * Self::VERTICAL_PREFERENCE) as u16
```

Reply:
the intention was to multiply floats to allow fractional preference setting if we need to tune this further. 

fixed here: https://github.com/trezor/trezor-firmware/pull/4060/commits/3c2ac84aebd094525372f20a3a6f6f087d3b9bd3